haystack/nodes/answer_generator/openai.py

```diff
@@ -292,7 +315,27 @@ def _fill_prompt(self, query: str, documents: List[Document]) -> str:
         ):
             kwargs["examples_context"] = self.examples_context
             kwargs["examples"] = example_prompts
-        full_prompt = next(self.prompt_template.fill(**kwargs))
+        prompt_template = self.prompt_template
+        # Switch for adding the prompt instructions at runtime.
+        if self.add_runtime_instructions:
+            temp = query.split("[SEPARATOR]")
+            if len(temp) != 2:
+                logger.error(
+                    "Instructions given to the OpenAIAnswerGenerator were not correct, please follow the structure "
```

Comment:
The instructions for OpenAIAnswerGenerator were incorrect. Make sure you follow the structure described in the docstrings (can we tell them explicitly which docstrings we mean here?)